File: python/moderation_text_token_demo.py
# -*- coding:utf-8 -*-
from moderation_sdk.gettoken import get_token
from moderation_sdk.moderation_text import moderation_text
from moderation_sdk.utils import init_global_env

if __name__ == '__main__':
    # Services currently support North China-Beijing(cn-north-4),China East-Shanghai1(cn-east-3), CN-Hong Kong(ap-southeast-1),AP-Singapore(ap-southeast-3)
    init_global_env('cn-north-4')

    #
    # access moderation text enhance,posy data by token
    #
    user_name = '******'
    password = '******'
    account_name = '******'  # the same as user_name in commonly use

    token = get_token(user_name, password, account_name)

    # call interface use the text
    result = moderation_text(token, '666666luo聊请+110亚砷酸钾六位qq，fuck666666666666666', 'content',
                             ['ad', 'politics', 'porn', 'abuse', 'contraband', 'flood'])
    print(result)

File: javascript/moderation_text_token_demo.js
/**
 *  图像内容检测服务token请求方式的使用示例
 */
var text = require("./moderation_sdk/moderation_text");
var token = require("./moderation_sdk/gettoken");
var utils = require("./moderation_sdk/utils");

// 初始化服务的区域信息，目前支持华北-北京(cn-north-4)、华东上海一(cn-east-3)、中国-香港(ap-southeast-1)、亚太-新加坡(ap-southeast-3)等区域信息
utils.initRegion("cn-north-4");

var username = "*******";        // 配置用户名
var domain_name = "*******";     // 配置用户名
var password = "*******";        // 密码

token.getToken(username, domain_name, password, function (token) {

    text.moderation_text(token, [{
            "text": "666666luo聊请+110亚砷酸钾六位qq，fuck666666666666666",
            "type": "content"}],
        ["ad", "abuse", "politics", "porn", "contraband"], function (result) {
            console.log(result);
        })

});
